cli/cmd/compliance.go

```diff
@@ -435,3 +441,49 @@ func statusToProperTypes(status string) string {
 		return "Unknown"
 	}
 }
+
+func validateRecommendationID(s string) bool {
+	match, _ := regexp.MatchString(RecommendationIDRegex, s)
+	return match
+}
+
+func outputResourcesByRecommendationID(report api.CloudComplianceReport) error {
+	violations := filterResourcesByRecommendationID(report, compCmdState.RecommendationID)
+
+	if len(violations) == 0 {
+		cli.OutputHuman("\nNo resources found affected by '%s'\n", compCmdState.RecommendationID)
+		return nil
+	}
+
+	if cli.JSONOutput() {
+		resourcesJsonOut := struct {
+			Violations []api.ComplianceViolation `json:"violations"`
+		}{violations}
+		return cli.OutputJSON(resourcesJsonOut)
+	}
+
+	cli.OutputHuman(
+		renderSimpleTable(
+			[]string{"Resource", "Region", "Reasons"},
```

Comment:
does the reason ever change? 🤔  If not, why do we print it multiple times.

Reply:
It does, it is possible to have the same violation on different resources for a different reason. It is just infrequent.